docs/advanced-usage.rst

```diff
@@ -127,15 +127,24 @@ You can connect to a proxy using HTTP, HTTPS or SOCKS. urllib3's behavior will
 be different depending on the type of proxy you selected and the destination
 you're contacting.
 
-When contacting a HTTP website through a HTTP or HTTPS proxy, the request will
-be forwarded with the `absolute URI
+When contacting a HTTP website through a HTTP proxy or a HTTPS proxy, the
```

Comment:
Maybe we can add two sections here, one for "HTTP Forwarding Proxies" and one for "HTTP Tunneling Proxies"?

Should also link to our SOCKS proxy documentation from here?

Also the link "When contacting a HTTP website through an HTTP proxy or HTTPS proxy" that's not correct is it? We should draw out a table comparing HTTP/HTTPS proxy w/ HTTP/HTTPS destination with and without `use_forwarding_for_https`. Make sure all prose aligns with this table.

Reply:
Good ideas! I made a couple of changes based on your feedback:

* I added sub-sections for SOCKS/HTTP/HTTPS.
* I clarified the behavior for all combinations of destination and proxy.

Let me know your thoughts!